pkg/cidata/cidata.TEMPLATE.d/boot/07-etc-environment.sh

```diff
@@ -4,22 +4,22 @@ set -eux
 # /etc/environment must be written after 04-persistent-data-volume.sh has run to
 # make sure the changes on a restart are applied to the persisted version.
 
+etc_environment_checksum_pre="nofile"
 if [ -e /etc/environment ]; then
+	etc_environment_checksum_pre="$(md5sum /etc/environment||printf 'forcepre')"
 	sed -i '/#LIMA-START/,/#LIMA-END/d' /etc/environment
 fi
 cat "${LIMA_CIDATA_MNT}/etc_environment" >>/etc/environment
+etc_environment_checksum_post="$(md5sum /etc/environment||printf 'forcepost')"
 
 # It is possible that a requirements script has started an ssh session before
 # /etc/environment was updated, so we need to kill it to make sure it will
 # restart with the updated environment before "linger" is being enabled.
 
-if command -v loginctl >/dev/null 2>&1; then
+if command -v loginctl >/dev/null 2>&1 && [ "${etc_environment_checksum_pre}" != "${etc_environment_checksum_post}" ]; then
```

Comment:
I'm not sure why we need to do the `md5sum` dance when we can just slurp the file into a variable. I also prefer somewhat shorter names.

Suggestion:

```
orig=$(test -f /etc/environment && cat /etc/environment)
if [ -f /etc/environment ]; then
	sed -i '/#LIMA-START/,/#LIMA-END/d' /etc/environment
fi
cat "${LIMA_CIDATA_MNT}/etc_environment" >>/etc/environment

# ...

if command -v loginctl >/dev/null 2>&1 && [ "${orig}" != "$(cat /etc/environment)" ]; then
```